packages/engine-server/src/workspace/service.ts

```diff
@@ -864,12 +866,33 @@ export class WorkspaceService implements Disposable, IWorkspaceService {
             return { repo, vaults, status: SyncActionStatus.NO_UPSTREAM };
           if (!(await this.shouldVaultsSync("pull", rootVaults)))
             return { repo, vaults, status: SyncActionStatus.SKIP_CONFIG };
-          if (await git.hasChanges({ untrackedFiles: "no" }))
-            return {
-              repo,
-              vaults,
-              status: SyncActionStatus.UNCOMMITTED_CHANGES,
-            };
+
+          let stashed: string | undefined;
+          if (await git.hasChanges({ untrackedFiles: "no" })) {
+            // stash away changes, otherwise pull will fail
+            try {
+              stashed = await git.stashCreate();
+              this.logger.info({ ctx, vaults, repo, stashed });
+              // this shouldn't fail, but for safety's sake
+              if (_.isEmpty(stashed) || !git.isValidStashCommit(stashed)) {
```

Comment:
something that is quite hard to do right now is debug, in a multi-vault, which vault failed. can we mention the vault name as part of the error?